pkg/kubelet/images/image_gc_manager.go

```diff
@@ -191,6 +190,10 @@ func (im *realImageGCManager) Start() {
 		if err != nil {
 			klog.Warningf("[imageGCManager] Failed to update image list: %v", err)
 		} else {
+			// The image list needs to be sorted when it gets read and used in
+			// setNodeStatusImages. We sort the list on write instead of on read,
+			// because the image cache is more often read than written
+			sort.Sort(sliceutils.ByImageSize(images))
```

Comment:
I was almost going to add another copy before this line... Perhaps it's better to put the copy and the sort inside `set()`, to make the code more contained? @liggitt WDYT?

Reply:
It looks like that's what used to happen:

https://github.com/kubernetes/kubernetes/blob/d5f1d5286fab4c730a5fbf8c38a07fab6ad2dba6/pkg/kubelet/images/image_gc_manager.go#L190-L194

copying/sorting on write makes sense to me if that statement about being read more frequently than written is true. Do we know why that was changed?

Reply:
moving copy and sort into set makes sense to me